astropy/time/formats.py

```diff
@@ -86,6 +86,11 @@ def __new__(mcls, name, bases, members):
         # it is not a user-accessible format and is only used for initialization into
         # a different format.
         if 'name' in members and cls.name != 'astropy_time':
+            # FIXME: check here that we're not introducing a collision with
+            # an existing method or attribute; problem is it could be either
+            # astropy.time.Time or astropy.time.TimeDelta, and at the point
```

Comment:
Isn't the problem collisions with method names in `TimeFormat`?

Reply:
Ah, no, I now see the problem from your example with `sort`. Though, I am surprised, the name `sort` can only be looked up via `__getattr__` and that supposedly happens *after* looking for methods like `sort()`. I'm puzzled...

Reply:
And looking further I see that indeed `t.sort()` works fine - it is the `t.format = 'sort'` that screws one up. As you write, tricky to debug!!